### PR TITLE
fix: @connection when field is given

### PR DESCRIPTION
**Pull request by akshbhu**
*Issue #5358 

*Description of changes:*

* added case for @coonection directive when field argument is present.

* added test for the same

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen-appsync-model-plugin/src/utils/process-connections.ts [hunk 1]
```diff
@@ -181,12 +182,24 @@ export function processConnections(
...
+            case 1:
+              return {
+                kind: CodeGenConnectionType.BELONGS_TO,
+                connectedModel: otherSide,
+                isConnectingFieldAutoCreated,
+                targetName: connectionFields[0] || makeConnectionAttributeName(model.name, field.name),
+              };
+            default:
+              throw new Error('DataStore only support one key in field');
+          }
```
Comment (yuth) [inline packages/amplify-codegen-appsync-model-plugin/src/utils/process-connections.ts:202]: The check should be based on if the field is nullable or non-nullable. When an field is non-nullable, it beconmes belogns to and when it is nullable it becomes `hasOne`

**Review by yuth — CHANGES_REQUESTED** [review]

File: packages/amplify-codegen-appsync-model-plugin/src/utils/process-connections.ts (hunk 1 above)
Comment (akshbhu) [reply packages/amplify-codegen-appsync-model-plugin/src/utils/process-connections.ts:202]: yes , thats true, But What about the case when @connection directive has a field which is non nullable. for example 
```graphql
          type Person @model {
            licenseId: ID!
            license: License @connection(fields: ["licenseId"])
          }
          type License @model {
            id: ID!
            holder: Person! @connection
          }
```
Here licence Id is non nullable . Wont this mean every licence(LicenceID) belongs to a person?

Comment (yuth) [reply packages/amplify-codegen-appsync-model-plugin/src/utils/process-connections.ts:202]: Logically doesn't it make sense. For a license to exists, there needs to be a person. So a person has a license and a license belongs to a person

**Comment by akshbhu** [comment]
Closing in favour of #6031

**Pull request closed without merging**